### PR TITLE
Test port forwarding feature

### PR DESCRIPTION
**Pull request by rakshith-ravi**
I kinda tried something out. Not entirely sure if I'm doing it right. I'm open to suggestions.

Do let me know if any changes are required.

Also, I loved the code and the way it's abstracted. Makes it super simple to use. Many thanks!

### CODE REVIEW

**Comment by rakshith-ravi** [comment]
PS, I haven't tested it yet. Let me know if this code structure is fine and I'll test it out.

File: src/port_forwarding.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            .arg(addr)
+            .spawn()
+            .map_err(Error::Ssh)?;
+        Ok(Self { child })
+    }
+}
+
+impl Drop for PortForward {
+    fn drop(&mut self) {
+        if let Some(ref mut stdin) = self.child.stdin {
```
Comment (jonhoo) [inline src/port_forwarding.rs:66]: I'm not sure I understand what this is needed for? Is the `kill` not sufficient? Does writing `exit` to stdin actually work?

**Review by jonhoo — COMMENTED** [review]
This looks really good over all! Would be good to test it though :p Maybe add an integration test too if you can think of one?

File: src/port_forwarding.rs (hunk 1 above)
Comment (rakshith-ravi) [reply src/port_forwarding.rs:66]: I just thought `kill` wouldn't be a graceful exit, so I thought sending `exit` to the `stdin` would be a more graceful way to quit the ssh session

Comment (jonhoo) [reply src/port_forwarding.rs:66]: Oh, yeah, I agree it's nicer, I'm just surprised that it works! After all, shouldn't `stdin` just be a path to the remote process? I guess in this instance there is no remote process, and that's why it works.

In any case, happy to merge this once it gets a test case!

**Comment by rakshith-ravi** [comment]
I'm developing on a mac so I'm not able to test properly. I'm pushing and expecting azure to test and give me some errors, sorry in advance for the spam xD

**Comment by rakshith-ravi** [comment]
Wait azure doesn't run the ssh based tests?

**Comment by jonhoo** [comment]
It should run the integration tests with a real ssh server running as well?

**Comment by rakshith-ravi** [comment]
How do I test integration tests on azure? Need some help on that pliss 🙈

**Comment by jonhoo** [comment]
They should run automatically as part of the CI. You can see the test failures on CI by clicking "details" on the failing test in the list on GitHub and then pressing the little "View more details on Azure Pipelines" link at the bottom, or just by going to https://dev.azure.com/jonhoo/jonhoo/_build?definitionId=23. I think Azure deletes test runs after not too many days though, so you may need to push a new commit to re-trigger CI so that you can see the actual failures for this one.

**Comment by jonhoo** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jonhoo/openssh-rs/23)

**Comment by NobodyXu** [comment]
@jonhoo I think this PR is superseded by #26 , which already added ssh forwarding support and integration tests.

**Pull request closed without merging**